casper/networks.py

```diff
@@ -0,0 +1,37 @@
+import random as r
+
+from casper.network import Network
+
+
+class SynchronousNetwork(Network):
+    def delay(self, sender, receiver):
+        return 0
+
+
+class StepNetwork(Network):
+    def delay(self, sender, receiver):
+        return 1
+
+
+class ConstantDelayNetwork(Network):
+    CONSTANT = 5
+
+    def delay(self, sender, receiver):
+        return self.CONSTANT
+
+
+class LinearDelayNetwork(Network):
+    MAX_DELAY = 5
+
+    def delay(self, sender, receiver):
+        return r.randint(1, self.MAX_DELAY)
```

Comment:
All of the networks specified above are synchronous (at least to us, as we know the delay :P ), as all of them has some max time that it takes a message to be delivered. 

I might rename SynchronousNetwork -> NoDelayNetwork, just so there's no confusing here. Also, StepNetwork is confusing to me - seems like just a variant of constant.

Reply:
👍 on NoDelayNetwork.

I like `StepNetwork` or something similarly named to cleanly expose it as a network to `casper.py`. It is the closest thing we have to how our network/simulations used to run so I like having it around.